src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -513,5 +520,15 @@ void GcsNodeManager::UpdatePlacementGroupLoad(
   });
 }
 
+void GcsNodeManager::AddDeadNodeToCache(std::shared_ptr<rpc::GcsNodeInfo> node) {
+  if (dead_nodes_.size() >= RayConfig::instance().maximum_gcs_dead_node_cached_count()) {
+    dead_nodes_.erase(sorted_dead_node_list_.begin()->first);
+    sorted_dead_node_list_.erase(sorted_dead_node_list_.begin());
+  }
+  auto node_id = NodeID::FromBinary(node->node_id());
+  dead_nodes_.emplace(node_id, node);
+  sorted_dead_node_list_.emplace_back(node_id, node->timestamp());
```

Comment:
Why don't we just use a priority queue instead of sorting only when it is restared?

Reply:
In the case of no restart, using list has less impact on performance.

Reply:
But priority queue also won't have performance impact right? It has O(logN) insertion/deletion, meaning with max 1000 entries, there will be almost no diff. 

Also, the current model seems like it will delete the wrong nodes if there's no restart (though I am not sure how common it will be)

Reply:
Each time the dead node is appended to the end of the list and the head of the list is the latest dead node. When the buffer is full, we can directly get the node ID from the list header node and delete it. We don't need extra overhead to sort. Sorting is required only when the GCS restarts and loads data from storage.

Reply:
Ah, I see. I neglected that appended node is always the oldest one. Thanks for the clarification :)!